mbp-with-mbo/src/main/java/com/epam/cme/mdp3/control/ChannelControllerRouter.java

```diff
@@ -131,14 +133,25 @@ protected void routeMBPEntry(int securityId, MdpMessage mdpMessage, MdpGroupEntr
         }
     }
 
+    protected void routeIncrementalComplete(Set<Integer> securityIds, MdpMessage mdpMessage, long msgSeqNum) {
+        for (Integer securityId : securityIds) {
+            InstrumentController instrumentController = instrumentManager.getInstrumentController(securityId);
+            if (instrumentController != null) {
+                instrumentController.handleIncrementalComplete(mdpMessage, msgSeqNum);
+            }
+        }
+    }
+    
     private void handleIncrementalMessage(MdpMessage mdpMessage, MdpGroup mdpGroup, MdpGroupEntry mdpGroupEntry, long msgSeqNum){
         if (isIncrementalMessageSupported(mdpMessage)) {
+            Set<Integer> securityIds = new HashSet<>();
```

Comment:
we can't create a new object every time we receive and handle a message it will lead to performance degradation and pauses

Reply:
true, could create at top level and clear